api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -288,6 +288,8 @@ func (m *AzureManagedControlPlane) Validate(cli client.Client) error {
 	allErrs = append(allErrs, validateNetworkDataplane(m.Spec.NetworkDataplane, m.Spec.NetworkPolicy, m.Spec.NetworkPluginMode, field.NewPath("spec").Child("NetworkDataplane"))...)
 
 	allErrs = append(allErrs, validateAPIServerAccessProfile(m.Spec.APIServerAccessProfile, field.NewPath("spec").Child("APIServerAccessProfile"))...)
+
+	allErrs = append(allErrs, validateAMCPVirtualNetwork(m.Spec, field.NewPath("spec").Child("VirtualNetwork"))...)
```

Comment:
Should we also be invoking this validation on AzureManagedControlPlaneTemplates?

Reply:
added via 6b1927121631e5ca59ef6dd42e4005af4ad506e1